recipes/heavydb/build-heavydb.sh

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env bash
+
+set -ex
+
+export EXTRA_CMAKE_OPTIONS="-GNinja"
+
+export INSTALL_BASE=opt/heavyai
+export BUILD_EXT=cpu
+export RUN_DBE_TESTS=1
+
+# Set flags
+case "$PKG_NAME" in
+
+    heavydb-common)
+        export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DENABLE_IMPORT_PARQUET=OFF"
+        ;;
+
+    heavydbe)
+        export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DENABLE_DBE=ON"
+        ;;
+
+    pyheavydbe)
+        export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DENABLE_DBE=ON"
+        ;;
+
+    heavydb)
+        # Make sure -fPIC is not in CXXFLAGS (that some conda packages may
+        # add), otherwise heavydb server will crash when executing generated
+        # machine code:
+        export CXXFLAGS="`echo $CXXFLAGS | sed 's/-fPIC//'`"
+
+        ;;
+
+    *)
+        echo "No specific flags set for $PKG_NAME"
+        ;;
+esac
+
+# Remove --as-needed to resolve undefined reference to `__vdso_clock_gettime@GLIBC_PRIVATE'
+export LDFLAGS="`echo $LDFLAGS | sed 's/-Wl,--as-needed//'`"
+
+export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DCMAKE_C_COMPILER=${CC} -DCMAKE_CXX_COMPILER=${CXX}"
+
+# Run tests labels:
+#   0 - disable building and running sanity tests
+#   1 - build and run the sanity tests
+#   2 - detect if sanity tests can be run, then set 1, otherwise set 0
+#
+# Ideally, this should be 2. However, the available conda-forge CI
+# disk space resourses are not sufficient for running the heavydb
+# sanity tests in full. Hence we disable the tests:
+export RUN_TESTS=0
+
+if [[ ! -z "${cuda_compiler_version+x}" && "${cuda_compiler_version}" != "None" && "$PKG_NAME" != "heavydb-common" ]]
+then
+    export BUILD_EXT=cuda
+    export RUN_DBE_TESTS=0
+    if [[ -z "${CUDA_HOME+x}" ]]
+    then
+        echo "cuda_compiler_version=${cuda_compiler_version} CUDA_HOME=$CUDA_HOME"
+        CUDA_GDB_EXECUTABLE=$(which cuda-gdb || exit 0)
+        if [[ -n "$CUDA_GDB_EXECUTABLE" ]]
+        then
+            CUDA_HOME=$(dirname $(dirname $CUDA_GDB_EXECUTABLE))
+        else
+            echo "Cannot determine CUDA_HOME: cuda-gdb not in PATH"
```

Comment:
This might not be available in some CUDA versions, iirc. Maybe it was only on Windows, though. But technically this is handled by `nvcc-feedstock` already.